packages/react-native-app-auth/index.d.ts

```diff
@@ -68,6 +68,7 @@ export type BaseAuthConfiguration = BaseConfiguration & {
 export type AuthConfiguration = BaseAuthConfiguration & {
   clientSecret?: string;
   scopes: string[];
+  claims: object,
```

Comment:
Yes, I considered this while writing the PR, but it does not quite work.

The best I can come up with is:

```suggestion
  claims?: {
    id_token?: Record<string, object | null>,
    userinfo?: Record<string, object | null>,
  }
```

This is because a Claims request can become quite complex, as described in Section 5.5 of the OIDC spec. The value of the `claims` key is "a JSON object listing the requested Claims", and each claim may have some more properties associated with it. I believe my suggested type here is even too restrictive, as the ODIC spec allows for more top level members, which then should be implementation-defined:

> Other members MAY be present. Any members used that are not understood MUST be ignored. 

Here is an example for a (complex) Claims request from the spec:

```json
  {
   "userinfo":
    {
     "given_name": {"essential": true},
     "nickname": null,
     "email": {"essential": true},
     "email_verified": {"essential": true},
     "picture": null,
     "http://example.info/claims/groups": null
    },
   "id_token":
    {
     "auth_time": {"essential": true},
     "acr": {"values": ["urn:mace:incommon:iap:silver"] }
    }
  }
```
 
It's up to you to decide how we want to type this, but either way, we should  make the key optional at the very least, otherwise this becomes a breaking change and is it also obviously not a required key.

```suggestion
  claims?: object,
```

Thanks for taking a look at this PR!